types/google-apps-script/google-apps-script-events.d.ts

```diff
@@ -22,12 +22,12 @@ declare namespace GoogleAppsScript {
       user: Base.User;
     }
 
-    interface AppsScriptHttpRequestEvent {
-      parameter: object;
+    interface AppsScriptHttpRequestEvent<T> {
+      parameter: { [key in keyof T]: string };
```

Comment:
I wonder is it need to use generics? IMO, key of parameter and parameters are always string.
Please show me why you use `T`.

```typescript
        interface AppsScriptHttpRequestEvent {
            parameter: { [key: string]: string };
            contextPath: string;
            contentLength: number;
            queryString: string;
            parameters: { [key: string]: string[] };
        }
```

Reply:
I used `T` because by binding the key of the object would reduce typos and other mistakes, both in the GAS project itself and in projects that share types.

Other patterns,
```ts
    interface DoGet<T = { [key: string]: string }> extends AppsScriptHttpRequestEvent<T> {
    }
```
By specifying the default type, I think it will be equivalent to the type you suggested if we don't use generics.

If it seems unnecessary, I will remove the generic and modify it to the suggested types.

Reply:
> In the above example, doGet(e) returns the following output:
https://developers.google.com/apps-script/guides/web#request_parameters

This example shows the type of `parameter` of doGet/doPost is always `{ [key: string]: string }` and type of `parameters` is `{ [key: string]: string[] }`.

Can you fix this pull request not to use generics?

Reply:
OK, and your suggestion seems very reasonable.
I will modify it in a way that does not use Generics.

Reply:
Thank you, @ivgtr.